pkg/services/rpcsrv/subscription_test.go

```diff
@@ -244,23 +244,23 @@ func TestFilteredSubscriptions(t *testing.T) {
 			},
 		},
 		"notification matching name": {
-			params: `["notification_from_execution", {"name":"my_pretty_notification"}]`,
+			params: `["notification_from_execution", {"name":"Transfer"}]`,
 			check: func(t *testing.T, resp *neorpc.Notification) {
 				rmap := resp.Payload[0].(map[string]any)
 				require.Equal(t, neorpc.NotificationEventID, resp.Event)
-				n := rmap["name"].(string)
-				require.Equal(t, "my_pretty_notification", n)
+				n := rmap["eventname"].(string)
+				require.Equal(t, "Transfer", n)
```

Comment:
The test is constructed improperly. We won't even notice if #3693 is repeated with some other notification. Refactor the test so that if `check` is not called when it's needed then the test should fail. It may be done by adding `shouldCheck bool` to the `cases` value struct.